src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/UsageBasedMetadataManager.cs

```diff
@@ -927,6 +927,11 @@ private void AddDataflowDependency(ref DependencyList dependencies, NodeFactory
                 methodILDefinition = FlowAnnotations.ILProvider.GetMethodIL(userMethod);
             }
 
+            // Data-flow (reflection scanning) for compiler-generated methods will happen as part of the
```

Comment:
I only understood what's going on after re-reading the pull request description - could you add a sentence describing when this can happen (the lines before this are supposed to do this conversion to user code).